src/radiation_system.hpp

```diff
@@ -1121,20 +1122,23 @@ void RadSystem<problem_t>::AddSourceTerms(array_t &consVar, arrayconst_t &radEne
 
 			// BEGIN NEWTON-RAPHSON LOOP
 			Egas_guess = Egas0;
-			EradVec_guess = Erad0Vec;
+			T_gas = quokka::EOS<problem_t>::ComputeTgasFromEint(rho, Egas_guess, massScalars);
+			AMREX_ASSERT(T_gas >= 0.);
+			fourPiB = chat * ComputeThermalRadiation(T_gas, radBoundaries_g_copy);
+			kappaPVec = ComputePlanckOpacity(rho, T_gas);
+			AMREX_ASSERT(!kappaPVec.hasnan());
+			// note that I assume kappa_P = kappa_E here
```

Comment:
I don't think we should assume $\kappa_P = \kappa_E$ for multigroup. This PR seems to require this to be the case.

Reply:
The algorithm that @chongchonghe and I designed should need to assume $\kappa_P = \kappa_E$, so I'm not sure why that comment is there. However, you still do need to compute $\kappa_P$ because that is the correct opacity to use to compute the matter emissivity. That is, the absorption term looks like $\kappa_E E$ for each group, but the emission term still looks like $\kappa_P(T_\mathrm{gas}) B(T_\mathrm{gas})$.

Reply:
See the updated description. Assuming this is for convenience, although we don't have to do that. It's safe to do so as of now because we always assume kappa_E = kappa_P in our code. In the future, we may want to do interpolation on the radiation energy and compute kappa_E based on the local radiation energy profile.

Reply:
Note that this assumption only affects the convergence rate of the Newton-Raphson iteration. It does not affect the result at all, even if kappa_P and kappa_E are not equally and their ratio depends on T. As a matter of fact, in all the tests where kappa depends on T (RadMarshakAsymptotic, RadhydroPulse, RadhydroPulseMG), you can set ComputePlanckOpacityTempDerivative to return 0.0 and the code would pass the tests without any problem.

Reply:
Ok, I think it makes sense now. I was also confused as to why the derivative of opacity w/r/t temperature dropped out.

I would strongly prefer that this is also documented in comments in the code, since we don't yet have a new code paper for this algorithm. Can you add your full description to the source code?

Reply:
Also, I think the current assumption that $\kappa_E = \kappa_P$ needs to be prominently highlighted in a comment at the start of this function.

Reply:
OK. I added back kappa_E and now kappa_E and kappa_P are allowed to differ. The only place where I assume kappa_E = kappa_P is at the calculation of the Jacobian. And again, this does not affect the converged result. 

I've also added sufficient comments to the code and made everything clear.